mirgecom/profiling.py

```diff
@@ -168,18 +149,57 @@ def _finish_profile_events(self) -> None:
 
             self.profile_results.setdefault(program, []).append(new)
 
-        for t in nonloopy_profile_events:
-            program = t.knl
-            time = t.cl_event.profile.end - t.cl_event.profile.start
-            new = ProfileResult(time, t.nops, t.nbytes, t.nbytes)
-            self.profile_results.setdefault(program, []).append(new)
-
         self.profile_events = []
-        nonloopy_profile_events = []
 
-    def tabulate_profiling_data(self) -> pytools.Table:
+    def get_profiling_data_for_kernel(self, kernel_name: str,
+                                 wait_for_events=True) -> ProfileResultsForKernel:
+        """Return value of profiling result for kernel `kernel_name`."""
```

Comment:
I'm not sure I like that this function resets the counters. At the very least, that behavior should be documented here.
```suggestion
    def get_and_reset_profiling_data_for_kernel(self, kernel_name: str,
                                 wait_for_events=True) -> ProfileResultsForKernel:
        """Return value of profiling result for kernel `kernel_name` and reset
        the internal profiling data for this kernel."""
```